sdk/identity/azure-identity/README.md

```diff
@@ -0,0 +1,186 @@
+|`ManagedIdentityCredential`|managed identity|constructor parameters
+|`EnvironmentCredential`|service principal|[environment variables](#environment-variables)
+|`ClientSecretCredential`|service principal|constructor parameters
+|`ClientCertificateCredential`|service principal|constructor parameters
+
+Credentials can be chained together to be tried in turn until one succeeds using the `ChainedTokenCredential`; see [chaining credentials](#chaining-credentials) for details.
+
+## DefaultAzureCredential
+`DefaultAzureCredential` is appropriate for most scenarios where the application is intended to run in the Azure Cloud. This is because the `DefaultAzureCredential` determines the appropriate credential type based of the environment it is executing in. It supports authenticating both as a service principal or managed identity, and can be configured so that it will work both in a local development environment or when deployed to the cloud. 
+
+The `DefaultAzureCredential` will first attempt to authenticate using credentials provided in the environment. In a development environment you can authenticate as a service principal with the `DefaultAzureCredential` by providing configuration in environment variables as described in the next section.
+
+If the environment configuration is not present or incomplete, the `DefaultAzureCredential` will then determine if a managed identity is available in the current environment.  Authenticating as a managed identity requires no configuration, but does
+require platform support. See the
+[managed identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/services-support-managed-identities) for more details on this.
+
+## Environment variables
+
+`DefaultAzureCredential` and `EnvironmentCredential` are configured for service
+principal authentication with these environment variables:
+
+|variable name|value
+|-|-
+|`AZURE_CLIENT_ID`|service principal's app id
+|`AZURE_TENANT_ID`|id of the principal's Azure Active Directory tenant
+|`AZURE_CLIENT_SECRET`|one of the service principal's client secrets
+
+# Examples
+
+## Authenticating with `DefaultAzureCredential`
+This example demonstrates authenticating the `SecretClient` from the [azure-keyvault-secrets][secrets_client_library] client library using the `DefaultAzureCredential`.
+```java
+// The default credential first checks environment variables for configuration as described above.
+// If environment configuration is incomplete, it will try managed identity.
+import com.azure.identity.credential.DefaultAzureCredential;
+import com.azure.security.keyvault.secrets.SecretClient;
+
+DefaultAzureCredential defaultCredential = new DefaultAzureCredential();
+
+// Azure SDK client builders accept the credential as a parameter
+
+SecretClient client = SecretClient.builder()
+    .endpoint("https://{YOUR_VAULT_NAME}.vault.azure.net")
+    .credential(credential)
+    .build();
```

Comment:
Use the 'new' API (that hasn't yet been merged).

Reply:
not part of this release.